### PR TITLE
Tweak Handlers::beginUnderOverlay and beginPages

### PR DESCRIPTION
**Pull request by m-holger**
No tests for getDictItems pending feedback.

### CODE REVIEW

File: libqpdf/JSON.cc [hunk 1]
```diff
@@ -410,6 +410,16 @@ JSON::isNull() const
     return m->value->type_code == vt_null;
 }
 
+JSON
+JSON::getDictItem(std::string const& key) const
+{
+    if (auto v = dynamic_cast<JSON_dictionary const*>(m->value.get())) {
+        if (auto it = v->members.find(key); it != v->members.end())
```
Comment (jberkenbilt) [inline libqpdf/JSON.cc:417]: Please put braces around the code in the condition.

**Review by jberkenbilt — CHANGES_REQUESTED** [review]
Thanks. I should have done it this way. I added the new code and almost did, but for some reason, I didn't.

File: libqpdf/JSON.cc (hunk 1 above)
Comment (m-holger) [reply libqpdf/JSON.cc]: Done. Also a couple of tests added.

**Review by jberkenbilt — APPROVED** [review]
Thanks. Will merge in spite of the extraneous blank lines.

**Pull request merged**